frontend/src/pages/answer/kgBubble/KgBubble.jsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable indent, no-use-before-define, func-names, no-return-assign */
+import React, {
+  useEffect, useRef, useContext,
+} from 'react';
+import * as d3 from 'd3';
+import Paper from '@material-ui/core/Paper';
+
+import BiolinkContext from '~/context/biolink';
+import kgUtils from '~/utils/knowledgeGraph';
+import dragUtils from '~/utils/d3/drag';
+import graphUtils from '~/utils/d3/graph';
+
+import './kgBubble.css';
+
+const nodePadding = 2;
+
+/**
+ * Knowledge Graph Bubble Graph
+ * @param {array} nodes - list of node objects
+ * @param {integer} numQgNodes - number of nodes in query graph
+ * @param {integer} numResults - number of results in message
+ */
+export default function KgBubble({
+  nodes, numQgNodes, numResults,
+}) {
+  const svgRef = useRef();
+  const { colorMap } = useContext(BiolinkContext);
+
+  /**
+   * Initialize the svg size
+   */
+  useEffect(() => {
+    if (nodes.length) {
+      const svg = d3.select(svgRef.current);
+      const { width, height } = svg.node().parentNode.getBoundingClientRect();
+      svg
+        .attr('width', width)
+        .attr('height', height)
+        .attr('preserveAspectRatio', 'xMinYMin meet')
+        .attr('viewBox', [0, 0, width, height]);
+    }
+  }, [nodes]);
+
+  function drawBubbleGraph() {
+    const svg = d3.select(svgRef.current);
+    const { width, height } = svg.node().parentNode.getBoundingClientRect();
+    // clear the graph
+    svg.selectAll('*').remove();
+    const getNodeRadius = kgUtils.getNodeRadius(width, height, numQgNodes, numResults);
+    const converted_nodes = nodes.map((d) => ({ ...d, x: Math.random() * width, y: Math.random() * height }));
```

Comment:
I wonder if it would make the subsequent simulation easier if we start with quasi-random positions rather than uniformly random ones. Something like a Halton sequence: https://en.wikipedia.org/wiki/Halton_sequence

This would also mean that the positions are in fact deterministic, which could be nice.

Reply:
Opened #193.